Use magic instead of make and port to mojo 25.3.0.dev2025032205

winding-lines opened the pull request:
This PR has 2 commits. The first one replaces make with magic. The second one makes the code compile and tests pass with nightly mojo.

The major mojo changes are:
- use `mut` instead of `inout`
- explicit constructors
- alignment is part of the type definition

winding-lines committed:
Move from makefile to magic project

Created a magic mojo project and migrated the Makefile and README to use
it.

winding-lines committed:
Make all the tests pass on mojo 25.3.0.dev2025032205

winding-lines committed:
Reformat the code

sa- commented:
Hey thanks for making the PR! I've enabled github actions on this PR but it looks like it's failing

winding-lines committed:
Change github workflow to use magic

Change the github workflow to use magic both for running the test and
the pre-commit.

winding-lines committed:
Add linux-64 as a magic platform

winding-lines committed:
Use full path to magic during github actions

sa- commented:
Thanks for the contribution! I'm sa-code on the discord server, feel free to DM me if you like

winding-lines commented:
Thanks for accepting my PR :) I am not sure where the discord server is?

sa- commented:
Ah on Modular's discord server. I also added you on linkedin (hopefully)
